plugins/woocommerce-blocks/tests/e2e/utils/create-dynamic-content.ts

```diff
@@ -0,0 +1,82 @@
+/**
+ * External dependencies
+ */
+import { readFile } from 'fs/promises';
+import Handlebars from 'handlebars';
+import { exec } from 'child_process';
+
+/**
+ * Internal dependencies
+ */
+import { cli } from './cli';
+
+export type GeneratedPost = {
+	url: string;
+	id: string;
+	deletePost: () => Promise< void >;
+};
+
+Handlebars.registerPartial(
+	'wp-block',
+	`
+<!-- wp:{{blockName}} {{{stringify attributes}}} -->
+{{> @partial-block }}
+<!-- /wp:{{blockName}} -->
+`
+);
+
+Handlebars.registerHelper( 'stringify', function ( context ) {
+	return JSON.stringify( context );
+} );
+
+const deletePost = async ( id: string ) => {
+	await cli(
+		`npm run wp-env run tests-cli -- wp post delete ${ id } --force`
+	);
+};
+
+// Creates a post as user 1.
+const createPost = async (
+	title: string,
+	templateContent: string
+): Promise< GeneratedPost > => {
```

Comment:
@dinhtungdu thats a good point, but I'm also not too concerned not to implement that, if someone else wants to extend this API they can do so easily when they need it?

I don't mind though, we certainly can pass an object representing more details about the post. I did wonder though, if we'd ever need it for this kind of testing. I guess if there were situations where our test was about how the block interacted with post meta and other attributes it might be relevant?

Reply:
I've only skimmed the framework here so far, but have we considered creating posts via a rest call? Like we do in gutenberg via RequestUtils fixture: https://github.com/WordPress/gutenberg/blob/trunk/packages/e2e-test-utils-playwright/src/request-utils/posts.ts#L60-L71

Reply:
@WunderBart we certainly could but wouldn't it be faster via cli? We already have access to the CLI installed on the docker box itself.

I don't mind the suggestion of course, just wondering what would be the benefit here of using the rest API directly?

Reply:
Actually I wonder if I'm making a bad assumption here. Maybe the rest API would perform better. certainly the code should be simpler.

Reply:
Good question. I don't know how it would perform speed-wise compared to the CLI, but I think that aiming for parity with the Gutenberg testing utils would be preferable in the long run. Since we're using wp-env as a testing env as well, we could even import the upstream [e2e testing utils](https://github.com/WordPress/gutenberg/tree/trunk/packages/e2e-test-utils-playwright) directly and use them OOTB.

I might, of course, be misunderstanding the intent here - I'm not even sure yet if we're already using the core e2e utils or not!

Reply:
> I might, of course, be misunderstanding the intent here - I'm not even sure yet if we're already using the core e2e utils or not!

@WunderBart you're good, I really appreciate your thoughts here and I agree after a minutes thought, matching with Gberg here makes sense and I can't imagine the perf being much different. I'll implement this. Thank you for this valuable suggestion. My biggest concern is authenticating the request because I don't know if these tests have prior art for that but i'm sure its solvable.

Thanks again its great feedback!

Reply:
> Maybe the rest API would perform better

@samueljseay  I believe it is. Using CLI we need some extra steps like spawning a node process and running the CLI command through `wp-env`. The later is painfully slow.

Reply:
> I'll implement this.

Fantastic! 🚀 

> My biggest concern is authenticating the request because I don't know if these tests have prior art for that but i'm sure its solvable.

Once we're logged in, the only thing that might be needed is the wp nonce attached to the request. I'd recommend checking against Gutenberg's `rest.ts` fixture module implementation - should be enough to make this work! 🤞 

If there's anything more I can assist with, just let me know!